dashboard/test/models/script_level_test.rb

```diff
@@ -681,45 +614,131 @@ class ScriptLevelTest < ActiveSupport::TestCase
   end
 
   test 'bonus levels do not appear in the normal progression' do
-    script = create :script
-    lesson_group = create :lesson_group, script: script
-    lesson = create :lesson, lesson_group: lesson_group, script: script
-    script_level = create :script_level, bonus: true, lesson: lesson, script: script
+    script_level = create_script_level_with_ancestors({bonus: true})
     assert_empty script_level.lesson.summarize[:levels]
   end
 
   test 'hidden_for_section returns true if lesson is hidden' do
-    script = create :script
-    lesson_group = create :lesson_group, script: script
-    lesson = create :lesson, lesson_group: lesson_group, script: script
-    script_level = create :script_level, script: script, lesson: lesson
+    script_level = create_script_level_with_ancestors
     section = create :section
 
-    create :section_hidden_lesson, lesson: lesson, section: section
+    create :section_hidden_lesson, lesson: script_level.lesson, section: section
 
     assert_equal true, script_level.hidden_for_section?(section.id)
   end
 
   test 'hidden_for_section returns true if script is hidden' do
-    script = create :script
-    lesson_group = create :lesson_group, script: script
-    lesson = create :lesson, lesson_group: lesson_group, script: script
-    script_level = create :script_level, script: script, lesson: lesson
+    script_level = create_script_level_with_ancestors
     section = create :section
 
-    create :section_hidden_script, script: script, section: section
+    create :section_hidden_script, script: script_level.script, section: section
 
     assert_equal true, script_level.hidden_for_section?(section.id)
   end
 
   test 'hidden_for_section returns false if no hidden lesson/script' do
+    script_level = create_script_level_with_ancestors
+    section = create :section
+
+    assert_equal false, script_level.hidden_for_section?(section.id)
+  end
+
+  test 'seeding_key without existing level_keys' do
```

Comment:
optional: what do you think about wrapping everything added here in another test case, like this? https://github.com/code-dot-org/code-dot-org/blob/bab074cc8c2e10ab5736711c56bd599c8f15379e/dashboard/test/models/unit_group_test.rb#L3-L6 it could help to clarify the scope of the helper methods, for example `create_seed_context` could be pulled inside but `create_script_level_with_ancestors` could be left outside.

Reply:
Seems like it'll take a bit of tinkering to be able to call the helper method from the enclosing class. I could probably make it a class method instead of instance method, but it seems like a small enough thing that I'd rather leave it.